File: ios_android/IFAWrapper/source/generic/IFAWrapper.cpp
/*
Generic implementation of the IFAWrapper extension.
This file should perform any platform-indepedentent functionality
(e.g. error checking) before calling platform-dependent implementations.
*/

/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */


#include "IFAWrapper_internal.h"
s3eResult IFAWrapperInit()
{
    //Add any generic initialisation code here
    return IFAWrapperInit_platform();
}

void IFAWrapperTerminate()
{
    //Add any generic termination code here
    IFAWrapperTerminate_platform();
}

const char* IFAGetAppleAdvertisingIdentifier()
{
	return IFAGetAppleAdvertisingIdentifier_platform();
}

bool IFAGetIsAdvertisingTrackingEnabled()
{
	return IFAGetIsAdvertisingTrackingEnabled_platform();
}

File: ios_android/IFAWrapper/source/android/IFAWrapper.java
/*
java implementation of the IFAWrapper extension.

Add android-specific functionality here.

These functions are called via JNI from native code.
*/
/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */
import com.ideaworks3d.marmalade.LoaderAPI;

class IFAWrapper
{
    public String IFAGetAppleAdvertisingIdentifier()
    {
        // not available in android
        
        return null;
    }
    public boolean IFAGetIsAdvertisingTrackingEnabled()
    {
        // not available in android
        
        return false;
    }
}
